### PR TITLE
Minify more JS

### PR DESCRIPTION
**Pull request by gravityrail**
Minifies all the JS files in _inc into _inc/build, with sourcemaps, and modifies existing code to point to those files.

This is factored out of our larger experimental #7963 branch.

### CODE REVIEW

**Comment by jeherve** [comment]
This would replace #5620

Fixes #3487

Related: #5425

File: gulpfile.babel.js [hunk 1]
```diff
@@ -75,6 +75,17 @@ function onBuild( done ) {
 				} );
 		}
 
+		// uglify some other random files
+		gulp.src( [ '_inc/*.js', '!_inc/*.min.js' ] )
+			.pipe( sourcemaps.init() )
```
Comment (zinigor) [inline gulpfile.babel.js]: Can we make sure we only generate sourcemaps when it's a development, and not a production build?

**Comment by ebinnion** [comment]
>When it's ready to ship we need to make sure we're not shipping unminified code anymore. 

This PR doesn't minify all files, so I don't think we're ready to do this project wide. But, I think we could ignore all non-minified files in `_inc`.

**Comment by ebinnion** [comment]
I tested several things such as Masterbar, Facebook embeds, etc., and they all seem to work.

@zinigor made a good point about not shipping sourcemaps in production and making sure we minimize shipping unminified scripts to production as well.

File: gulpfile.babel.js (hunk 1 above)
Comment (gravityrail) [reply gulpfile.babel.js]: So should we also eliminate all the CSS .map files too? We already output those for production builds.

**Comment by gravityrail** [comment]
@zinigor none of the files being minified are already minified.

I have added code which only compiles sourcemaps if the environment is not production.

I think this is good to go.

**Comment by ebinnion** [comment]
> @zinigor none of the files being minified are already minified.

I believe what @zinigor is suggesting, is that since we're going to start shipping minified files for the above, we should not also ship the unminified versions of those files, which will help reduce the download size a tad.

**Comment by ebinnion** [comment]
To test:

- Checkout PR
- `yarn build`
- Test various parts of Jetpack
    - Test IDC by triggering an IDC and ensuring the notice shows
    - Trigger connection banner by deactivating and then reactivating Jetpack
    - Visit `$site.com/wp-admin/admin.php?page=jetpack_modules` to trigger the modules js files
    - Embed a Facebook post
    - etc.

Further, I also commented out a bit of the logic in `deploy-to-master-stable.sh` script so that it didn't actually commit to Git so that I could test the build. And it seemed to remove map files as expected as well

**Comment by ebinnion** [comment]
After rebasing against master again, everything is now working as expected.

Testing instructions are in the previous comment.

**Comment by gravityrail** [comment]
LGTM! 👍

**Comment by enejb** [comment]
I ❤️ this! Thanks for working on it. 
@zinigor I think we should always ship the full versions of the js and css files for human readablity. 
As per plugin item #4 in the plugin guidelines. Which suggests 'Minified code may be used, however the unminified versions should be included whenever possible. ' 
https://developer.wordpress.org/plugins/wordpress-org/detailed-plugin-guidelines/

Or maybe just need to include a comment where to find the full human readable version of the code.
Also if we are shipping the full versions it would be good to add a conditional that checks for the SCRIPT_DEBUG and serves the full version when it is present.

**Comment by ebinnion** [comment]
@oskosk and I chatted a bit about this in Slack today and I think we're going to move forward with shipping the minified and non-minified versions, and using `SCRIPT_DEBUG` to determine which one to load.

**Comment by ebinnion** [comment]
I'm currently working on a similar PR to start addressing modules in #8153.

**Pull request merged**